README.md

```diff
@@ -31,65 +26,153 @@ A set of React hooks for React Native Navigation
     </tr>
 </table>
 
-## Usage
+## Installation
+
+- `npm install react-native-navigation-hooks --save`, or
+- `yarn add react-native-navigation-hooks`
+
+## Hooks
+
+### `useNavigation`
+
+This hook provides you a set of navigation helpers that aims to reduce the boilerplate required to perform common navigation actions specially those that need the `componentId` argument.
+
+In order to use this hook your screens must use the `<NavigationProvider>` to pass the `componentId` to the three below so the `useNavigation` hook can read it.
+
+You can take advantage of the [withNavigationProvider](#withNavigationProvider) or you could also use the `<NavigationProvider>` manually.
+
+```tsx
+Navigation.registerComponent('MyScreenComponent', () => withNavigationProvider(MyScreenComponent))
+
+// or
+
+import { ComponentIdContext } from 'react-native-navigation-hooks'
```

Comment:
This should be `NavigationProvider`,  right?

Reply:
fixed on 0eb2812117965941ade082a6b7300f236054a207